src/invertedindex.js

```diff
@@ -0,0 +1,222 @@
+
+const hasProperty = Object.prototype.hasOwnProperty;
+let instance = null;
+
+class InvertedIndex {
+  /**
+    * constructor method
+  */
+  constructor() {
+    if (!instance) {
+      instance = this;
+      this.filesIndexed = {};
+      this.inputData = {};
+      this.error = {};
+    }
+    return instance;
+  }
+
+  /**
+   * readFile function is used to get all the index
+   * @param {object} inputData - the json data to index
+   * @return {reject(false)} false - When file is of bad extent of
+   * invalid json format
+   * @return {resolve(true)} true - When file is of the right extension structure
+   */
+  readFile(inputData) {
+    return new Promise((resolve, reject) => {
+      if (!inputData.name.match(/\.json$/)) {
+        return reject(false);
+      }
+      const readFile = new FileReader();
+      readFile.readAsText(inputData);
+      readFile.onload = (file) => {
+        const content = file.target.result;
+        try {
+          return resolve(JSON.parse(content));
+        } catch (exception) {
+          return reject(false);
+        }
+      };
+    });
+  }
+
+  /**
+   * prepareJsonData gets the json ready for indexing by tokenizing statements
+   * @param {object} inputData - the json data to index
+   * @param {string} filename - the name of the file to be indexed
+   * @return {boolean} true or false if the createIndex was successful
+  */
+  createIndex(inputData, filename) {
```

Comment:
It is always good practice to keep methods as short and as specific as possible. You can extract line 56-59 into a function and reuse at line 63-66. The function may look like this.
```
handleError(fileName, errorMessage, ) {
    delete this.filesIndexed[fileName];
    this.error.status = true;
    this.error.message = errorMessage;
    throw this.error;
}
```
You can now have something like
```
try {
     if (Object.keys(inputData).length < 1) {
          this.handleError(filename, 'File contains no document');
     }
     Object.keys(inputData).forEach((eachIndex) => {
          if (!this.validateFile(inputData[eachIndex])) {
              this.handleError(filename, 'Incorrect Document Structure');
         }
         words.push(this.getDocumentTokens(inputData, documentNum));
         documentNum += 1;
     });
```
This appears neater.